### PR TITLE
Use timeout from context

### PR DESCRIPTION
**Pull request by lucasmdrs**
This PR bump a few deps and remove timeout from GitHub HTTP client in favor of context.

### CODE REVIEW

File: github.go [hunk 1]
```diff
@@ -66,19 +65,13 @@ func NewGithubClient(
 	token string,
 	releaseFilter ReleaseFilter,
 	assetFilter AssetFilter,
-	connectionTimeout time.Duration,
```
Comment (vgarvardt) [inline github.go:69]: with this change we should either bump major version as it breaks BC, or store this and create timeout context internally - not sure what is better in this case

Comment (lucasmdrs) [reply github.go:69]: I will make some changes to keep compatibility for now :+1:  
but we might need to do break this in the future, currently it accepts no token but Github V4 API [requires the requests to be authenticated](https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#authenticating-with-graphql) to perform GraphQL operation

Comment (vgarvardt) [reply github.go:69]: Create a ticket not to forget about it. OR maybe do it right away

File: github.go [hunk 2]
```diff
@@ -69,30 +70,32 @@ func NewGithubClient(
 	connectionTimeout time.Duration,
 ) *GithubLocator {
 	var httpClient *http.Client
+	ctx, cancel := ensureTimeout(ctx, connectionTimeout)
```
Comment (vgarvardt) [inline github.go:73]: if there is more than 10s from the client creation to the first request - it will already be cancelled

File: github.go [hunk 3]
```diff
@@ -69,30 +70,32 @@ func NewGithubClient(
...
+
 	if token != "" {
 		src := oauth2.StaticTokenSource(
 			&oauth2.Token{AccessToken: token},
 		)
 		httpClient = oauth2.NewClient(ctx, src)
-		httpClient.Timeout = connectionTimeout
-	}
-
-	if httpClient == nil {
```
Comment (vgarvardt) [inline github.go:80]: if token is not set - we'll get `nil` client

Comment (lucasmdrs) [reply github.go:80]: This is fine, the github client constructor checks if the httpClient is nil and uses the default one

Comment (lucasmdrs) [reply github.go:80]:
```go
	if httpClient == nil {
		httpClient = http.DefaultClient
	}
```

Comment (vgarvardt) [reply github.go:80]: default client comes w/out any timeout

Comment (lucasmdrs) [reply github.go:80]: the timeout is controller by context, not on the http client

**Review by vgarvardt — APPROVED** [review]

**Pull request merged**